skfp/distances/tanimoto.py

```diff
@@ -0,0 +1,317 @@
+    ----------
+    .. [1] `Bajusz, D., Rácz, A. & Héberger, K.
+       "Why is Tanimoto index an appropriate choice for fingerprint-based similarity calculations?"
+       J Cheminform, 7, 20 (2015).
+       <https://jcheminf.biomedcentral.com/articles/10.1186/s13321-015-0069-3>`_
+
+    Examples
+    ----------
+    >>> from skfp.similarities import tanimoto_count_similarity
+    >>> vec_a = np.array([7, 1, 1])
+    >>> vec_b = np.array([7, 1, 2])
+    >>> sim = tanimoto_count_similarity(vec_a, vec_b)
+    >>> sim
+    0.98
+
+    >>> vec_a = csr_array(([7, 1, 1]))
+    >>> vec_b = csr_array(([7, 1, 2]))
+    >>> sim = tanimoto_count_similarity(vec_a, vec_b)
+    >>> sim
+    0.98
+    """
+    _check_nan(vec_a)
+    _check_nan(vec_b)
+
+    if np.sum(vec_a) == 0 and np.sum(vec_b) == 0:
+        return 1.0
+
+    if isinstance(vec_a, csr_array) and isinstance(vec_b, csr_array):
+        return _tanimoto_count_scipy(vec_a, vec_b)
+    elif isinstance(vec_a, np.ndarray) and isinstance(vec_b, np.ndarray):
+        return _tanimoto_count_numpy(vec_a, vec_b)
+    else:
+        raise TypeError(
+            f"Both vec_a and vec_b must be of the same type: either numpy.ndarray "
+            f"or scipy.sparse.csr_array, got {type(vec_a)} and {type(vec_b)}"
+        )
+
+
+@validate_params(
+    {
+        "vec_a": ["array-like", csr_array],
+        "vec_b": ["array-like", csr_array],
+    },
+    prefer_skip_nested_validation=True,
+)
+def tanimoto_count_distance(
+    vec_a: Union[np.ndarray, csr_array], vec_b: Union[np.ndarray, csr_array]
+) -> float:
+    """
+    Computes the Tanimoto distance for binary data between two input arrays or sparse matrices
+    by subtracting similarity value from 1.
+    Calculated distance falls within the range from 0 to 1.
+    Passing all-zero vectors to this function results in distance of 0.
+
+    Parameters
+    ----------
+    vec_a : {ndarray, sparse matrix}
+        First binary input array or sparse matrix.
+
+    vec_b : {ndarray, sparse matrix}
+        Second binary input array or sparse matrix.
+
+    Returns
+    ----------
+    distance : float
+        Tanimoto distance between vec_a and vec_b.
+
+    Examples
+    ----------
+    >>> from skfp.similarities import tanimoto_count_distance
+    >>> vec_a = np.array([7, 1, 1])
+    >>> vec_b = np.array([7, 1, 2])
+    >>> dist = tanimoto_count_distance(vec_a, vec_b)
+    >>> dist
+    0.02
+
+    >>> vec_a = csr_array(([7, 1, 1]))
+    >>> vec_b = csr_array(([7, 1, 2]))
+    >>> dist = tanimoto_count_distance(vec_a, vec_b)
+    >>> dist
+    0.02
+    """
+
+    return 1 - tanimoto_count_similarity(vec_a, vec_b)
+
+
+@njit(parallel=True)
+def _tanimoto_count_numpy(vec_a: np.ndarray, vec_b: np.ndarray) -> float:
+    """
+    Calculates the Tanimoto similarity between two count data numpy arrays.
+    This function uses Numba for JIT compilation and parallel computations to
+    efficiently compute the similarity between two vectors.
+    """
```

Comment:
I don't think that docstring is necessary here, the function is very readable already